### ISSUE TITLE
[Question] playwright.tester: How to wait on webServer.command

### ISSUE THREAD

**Issue by stavros-mobal** [microsoft/playwright#11811]
I am using the following configuration for starting the e2e-server:
```
   webServer: {
        /* To see the stdout, you can set the DEBUG=pw:webserver environment variable. */
        command: `vue-cli-service serve --port ${WEBSERVER_PORT} --mode production`,
        port: WEBSERVER_PORT,
        timeout: 120 * 1000,
        reuseExistingServer: !process.env.CI,
    },
```

However, it seems to me that tests start before the serving is complete.

As a POC, I have one spec that depends on `vue-cli-service serve` starting, and one that does not.
I would be fine if the non-dependant tests started earlier; not with the dependent tests, however.

A different plugin seems to have access to vue internals (https://github.com/marcol/vue-cli-plugin-e2e-playwright/blob/da680ebbc97d429f660d9bade626580cb833da40/index.js#L20-L21)
It would be nice if Playwright tester could get the same functionality - instead of depending on the port being "detected as used".

**Comment by yury-s** [comment]
Playwrfignt [will wait](https://playwright.dev/docs/api/class-testconfig#test-config-web-server) for the server to start listening on the configured port before running the tests. In the current version if that's still too early you can put the logic that waits for the server to start serving into `beforeAll()` for the tests that depend on it.

https://github.com/microsoft/playwright/pull/10138 added [`url`](https://playwright.dev/docs/next/api/class-testconfig#test-config-web-server) to the server configuration and in the @next playwright version you can specify a url to have playwright wait until it returns OK status. This is going to be published in 1.19

**Comment by mxschmitt** [comment]
I'll reopen for now until I verified manually that it works as expected.

**Comment by mxschmitt** [comment]
I tried to create a new Vue 3 application with their CLI but for me it was working as expected. @stavros-mobal would it be possible that you can create a repro for us? Thank you!

**Comment by stavros-mobal** [comment]
Vue2 _not_ Vue3 ;-)

**Comment by mxschmitt** [comment]
I tried to reproduce with Vue 2, it is still working for me. Could you provide us a repro with your configuration so we can try locally?

**Comment by stavros-mobal** [comment]
My repro was local; brute-forcing it seemed to work (but that could mean that the "local state" was altered). The goal (and what I am building) is a Github Action.

Whether I replicate this or not, however, I am not sure if I can share / what can I share: the code is private 😕

**Comment by zomars** [comment]
Also it seems that globalSetup runs before webServer is started.

**Comment by mxschmitt** [comment]
Closing since we can't repro.

@zomars please file a separate issue.
